src/V3SchedTiming.cpp

```diff
@@ -309,41 +310,53 @@ void transformForks(AstNetlist* const netlistp) {
             iterateChildren(nodep);
             m_funcp = nullptr;
         }
-        virtual void visit(AstVar* nodep) override { nodep->user1(true); }
+        virtual void visit(AstVar* nodep) override {
+            if (!m_forkp) nodep->user1(true);
+        }
         virtual void visit(AstFork* nodep) override {
-            VL_RESTORER(m_forkp);
+            if (m_forkp) return;  // Handle forks in forks after moving them to new functions
             m_forkp = nodep;
```

Comment:
```suggestion
            if (m_forkp) return;  // Handle forks in forks after moving them to new functions
            VL_RESTORER(m_forkp);
            m_forkp = nodep;
```
Then remove setting to nullptr at bottom.
I think this still works, and is easier to avoid future problems.

Reply:
Done